### DISCUSSION TITLE
Revealer reverse leaves widget

### DISCUSSION THREAD

**Discussion by TornaxO7** [elkowar/eww#575] [Q&A] [answered]
Hello! Somehow the revealer doesn't clean up its path after revealing its children.

## Demo

https://user-images.githubusercontent.com/50843046/190467727-58cb2d17-5fe4-4a7f-a8ca-4e9421060106.mp4

## Expected
I expected to get back to the following state after un-revealing the children:

![image](https://user-images.githubusercontent.com/50843046/190467830-e0b403cb-64e6-4f6d-9e8b-b7ed6c9108e3.png)

## Example code (used in video)
`~/.config/eww/eww.yuck`

<details>

```yuck
(defvar show_example false)

(defwindow yeet
  :monitor 0
  :windowtype "toolbar"
  :wm-ignore true
  :stacking "bg"
  :geometry (geometry :x "-10%"
              :y "-10%"
              :width "1%"
              :height "1%"
              :anchor "bottom right")
  (box
    :orientation "horizontal"
    :space-evenly false
    :spacing 1
    (button
      :onclick "./toggle_example.sh"
      (label :text "click me!"))
    (revealer
      :transition "slideleft"
      :reveal show_example
      :duration "1s"
      (box
        (label :text "entry 1")
        (label :text "entry 2")))))
```

</details>

`~/.config/eww/toggle_example.sh`

<details>

```bash
#!/bin/bash
state=$(eww get show_example)

if [[ $state == "true" ]]
then
    eww update show_example=false
else
    eww update show_example=true
fi
```

</details>

What am I doing wrong?

**Comment by viandoxdev** [comment, marked as answer]
If the gray is the color of the background itself then this is a known limitation, gtk windows never shrink (see #196 a bit long tho so TL;DR gtk problem). You can work around that by using a transparent background window that always occupies the full space and have all your content in a revealer
